### PR TITLE
Reuse fetching/error display logic in console views

### PR DESCRIPTION
**Pull request by bafonins**
#### Summary

Closes https://github.com/TheThingsNetwork/lorawan-stack/issues/1061

#### Changes

- Start application/gateway stream only if the application/gateway exists
- Adjust all views mentioned in the issue

### CODE REVIEW

File: pkg/webui/console/store/middleware/logics/applications.js [hunk 1]
```diff
@@ -25,9 +25,11 @@ import createEventsConnectLogics from './events'
 
 const getApplicationLogic = createRequestLogic({
   type: applications.GET_APP,
-  async process ({ action }) {
+  async process ({ action }, dispatch) {
     const { payload: { id }, meta: { selector }} = action
-    return api.application.get(id, selector)
+    const app = await api.application.get(id, selector)
```
Comment (kschiffer) [inline pkg/webui/console/store/middleware/logics/applications.js]: This doesn't seem right to me; adding the dispatch here mixes responsibilities. Can we not solve this by orchestrating the dispatches in the component (e.g. using the promise and async/await)?

Comment (bafonins) [reply pkg/webui/console/store/middleware/logics/applications.js]: https://github.com/TheThingsNetwork/lorawan-stack/commit/ebdc6b422eceda37a30b06c850d8f00fca65ebbd#diff-b88e01b98363d1c7259c98f3dd34cf7b

**Comment by bafonins** [comment]
> Could you also take care of this warning then

No I cant, this has nothing to do with this pr. Ill create a `quickfix` pr for this.

**Comment by kschiffer** [comment]
What did https://github.com/TheThingsNetwork/lorawan-stack/pull/1064/commits/caad4e8eb073b7f1b23a0b9fbeae1cea9d23ea21 have to do with this PR then?

**Comment by bafonins** [comment]
absolutely nothing. and it is also your responsibility as a reviewer to point this out before approving

**Pull request merged**